File: Data Structures & Algorithms/daily-temperatures/submission-3.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temp) {
      
      vector<int> res(temp.size(),0);
      stack<pair<int,int>> s;

      for(int i=0;i<temp.size();i++){
         int currTemp = temp[i];
         while(!s.empty() && currTemp>s.top().first){
            auto [temp,index] = s.top();
            s.pop();
            res[index] = i-index;
         }
         s.push({currTemp,i});
      }

      return res;
    }
};

File: Data Structures & Algorithms/climbing-stairs/submission-3.py
class Solution:
    def climbStairs(self, n: int) -> int:
        
        if n < 2:
            return 1
        
        first = 1
        second = 1

        for _ in range(2, n+1):
            temp = second
            second = first + second
            first = temp
        return second